src/object/cli_obj.c

```diff
@@ -1957,6 +2076,53 @@ obj_ec_recov_cb(tse_task_t *task, struct dc_object *obj,
 		D_ERROR("task %p "DF_OID" EC recovery failed "DF_RC"\n",
 			task, DP_OID(obj->cob_md.omd_id), DP_RC(rc));
 	}
+	return rc;
+}
+
+static int
+obj_ec_check_exist(tse_task_t *task, struct dc_object *obj, struct obj_auxi_args *obj_auxi)
+{
+	daos_obj_fetch_t		*args = dc_task_get_args(task);
+	tse_task_t			*check_task;
+	int				 rc;
+
+	rc = dc_obj_fetch_task_create(args->oh, obj_auxi->th, args->flags, args->dkey, args->nr,
+				      DIOF_CHECK_EXISTENCE, args->iods, NULL, NULL, NULL, NULL,
+				      NULL, tse_task2sched(task), &check_task);
```

Comment:
Hmm, could you please explain why create a new fetch task?

Reply:
agreed that can achieve the same result if reinit the original task, but need to change some parameters of original task like the extra flag of DIOF_CHECK_EXISTENCE, and change the sgl to be NULL, that need to set it back when finish the original task. Also with a few extra detailed code, so I just create a new sub-task and need not change original task's parameter. 
BTW, creating DIOF_CHECK_EXISTENCE sub-task also used by fanyong's dtx code, so here seems fine to use the same kind of handling.

Reply:
Sorry, actually I mean why can not we use obj_retry_cb()?  Sorry I really hate these special flags, causing a lot problems for fixing bugs and maintenance.

> DIOF_CHECK_EXISTENCE, and change the sgl to be NULL, that need to set it back when finish the original task. 

sorry, is this the only difference? other seems only for validation checking?

Reply:
no problem, I also hate those flags, but cannot fully avoid for some cases. BTW, the DIOF_CHECK_EXISTENCE is not added by this patch, already existed on master code.
I changed the code to reuse obj_retry_cb(), and remove one flag - wait_check_exist, myself testing seems work fine.